pkg/specgen/generate/container_create.go

```diff
@@ -154,6 +166,16 @@ func MakeContainer(ctx context.Context, rt *libpod.Runtime, s *specgen.SpecGener
 		logrus.Debugf("setting container name %s", s.Name)
 		options = append(options, libpod.WithName(s.Name))
 	}
+	if len(s.DevicesFrom) > 0 {
+		for _, dev := range s.DevicesFrom {
+			ctr, err := rt.GetContainer(dev)
+			if err != nil {
+				return nil, nil, nil, err
+			}
+			devices := ctr.Config().DeviceHostSrc
```

Comment:
Same here, please add a public accessor for DeviceHostSrc ,e.g. `ctr.GetDeviceHostSrc()`
`ctr.Config()` also uses `JSONDeepCopy`